saleor/graphql/discount/dataloaders.py

```diff
@@ -150,6 +151,57 @@ def batch_load(self, keys):
         ]
 
 
+class VoucherInfoByVoucherCodeLoader(DataLoader):
+    context_key = "voucher_info_by_voucher_code"
+
+    def batch_load(self, keys):
+        vouchers_map = (
+            Voucher.objects.using(self.database_connection_name)
+            .filter(code__in=keys)
+            .in_bulk(field_name="code")
+        )
+        vouchers = vouchers_map.values()
+        voucher_products = Voucher.products.through.objects.filter(
+            voucher__in=vouchers
+        ).values_list("voucher_id", "product_id")
+        voucher_variants = Voucher.variants.through.objects.filter(
+            voucher__in=vouchers
+        ).values_list("voucher_id", "productvariant_id")
+        voucher_collections = Voucher.collections.through.objects.filter(
+            voucher__in=vouchers
+        ).values_list("voucher_id", "collection_id")
+        voucher_categories = Voucher.categories.through.objects.filter(
+            voucher__in=vouchers
+        ).values_list("voucher_id", "category_id")
```

Comment:
We should add `.using(self.database_connection_name)` between `objects` and `filter`.